src/clowder.ts

```diff
@@ -212,6 +212,7 @@ export class Clowder {
     method,
     itemSelect,
     randomlySelectUnvalidated = false,
+    returnUndefinedOnExhaustion = false, // New parameter
```

Comment:
Can we make the default value for this parameter `true`. I know it will be a hassle in terms of rewriting some of the tests. But I think that Anya wanted the default behavior to be `true`. And I think it makes more sense that way as well.

Reply:
Sure! yes, make sense